sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -1,142 +1,148 @@
+Familiarity with `azure-eventhubs` and `azure-eventhubs-eph` package is assumed. For those new to the Event Hubs client
+library for Java, please refer to the [README.md][README] rather than this guide.
 
 ## Table of contents
 
 - [Table of contents](#table-of-contents)
-- [Prerequisites](#prerequisites)
-- [Updated Maven dependencies](#updated-maven-dependencies)
-- [General changes](#general-changes)
-  - [Converting core classes](#converting-core-classes)
-  - [Sending events](#sending-events)
-  - [Receiving events](#receiving-events)
-  - [Minor renames](#minor-renames)
-- [Migration samples](#migration-samples)
-  - [Migrating code from `PartitionSender` to `EventHubProducerAsyncClient` for sending events to a partition](#migrating-code-from-partitionsender-to-eventhubproducerasyncclient-for-sending-events-to-a-partition)
-  - [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events using automatic routing](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-using-automatic-routing)
-  - [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events with partition key](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-with-partition-key)
-  - [Migrating code from `PartitionReceiver` to `EventHubConsumerAsyncClient` for receiving events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerasyncclient-for-receiving-events-in-batches)
-  - [Migrating code from `EventProcessorHost` to `EventProcessorClient` for receiving events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-receiving-events)
+- [Migration benefits](#migration-benefits)
+    - [Cross Service SDK improvements](#cross-service-sdk-improvements)
+- [Important changes](#important-changes)
+    - [Package names and namespaces](#package-names-and-namespaces)
+    - [Client hierarchy and constructors](#client-hierarchy-and-constructors)
+        - [Create an Event Hub producer](#create-an-event-hub-producer)
+    - [Send events to a specific Event Hub partition](#send-events-to-a-specific-event-hub-partition)
+    - [Send events using automatic routing](#send-events-using-automatic-routing)
+    - [Receive events from a single partition](#receive-events-from-a-single-partition)
+    - [Receiving events from all partitions](#receiving-events-from-all-partitions)
+        - [V3 Checkpoints](#v3-checkpoints)
 - [Additional samples](#additional-samples)
 
-## Prerequisites
-Java Development Kit (JDK) with version 8 or above.
+## Migration benefits
 
-## Updated Maven dependencies
+A natural question to ask when considering whether or not to adopt a new version or library is the benefits of doing so. As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Java
```

Comment:
`is the benefits of doing so` - > `is what the benefits of doing so`

Reply:
Eh, Grammarly checked this paragraph.